lib/tapjoy/ldap/api/user.rb

```diff
@@ -0,0 +1,114 @@
+require 'memoist'
+module Tapjoy
+  module LDAP
+    module API
+      module User
+        class << self
+          extend Memoist
+          def create(fname, lname, type, group)
+            # Properly capitalize names
+            fname, lname = [fname, lname].map(&:titleize)
+
+            Tapjoy::LDAP::client.add(
+              distinguished_name(fname, lname, type),
+              ldap_attr(fname, lname, type, group)
+            )
+          end
+
+          def destroy(username, type)
+            Tapjoy::LDAP::client.delete(
+              distinguished_name(*name_of_user(username), type)
+            )
+          end
+
+          def index
+            Tapjoy::LDAP::client.search('*', filter(uid: '*'))
+          end
+
+          def show(username)
+            Tapjoy::LDAP::client.search('*', filter(uid: username))
+          end
+
+          private
+
+          # Filter users for #show and #index
+          def filter(uid: '*')
+            Net::LDAP::Filter.eq('uid', uid)
+          end
+          memoize :filter
```

Comment:
This may pose an issue when used in the self_service application: the memoization is at the Class level and calls will be cached until the app restarts.